core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1473,6 +1484,30 @@ class ReplicaManager(val config: KafkaConfig,
     }
   }
 
+  /**
+   * Checks if the topic ID provided in the LeaderAndIsr request is consistent with the topic ID in the log.
+   *
+   * If the request had an invalid topic ID (null or zero), then we assume that topic IDs are not supported.
+   * The topic ID was not inconsistent, so return true.
+   * If the log does not exist or the topic ID is not yet set, logTopicIdOpt will be None.
+   * In both cases, the ID is not inconsistent so return true.
+   *
+   * @param requestTopicId the topic ID from the LeaderAndIsr request
+   * @param logTopicIdOpt the topic ID in the log if the log and the topic ID exist
+   * @return true if the request topic id is consistent, false otherwise
+   */
+  private def checkTopicId(requestTopicId: Uuid, logTopicIdOpt: Option[Uuid]): Boolean = {
+    if (requestTopicId == null || requestTopicId == Uuid.ZERO_UUID) {
+      true
+    } else
+      logTopicIdOpt match {
```

Comment:
You can use `logTopicIdOpt.contains(requestTopicId)`

Reply:
I've changed the match portion to `logTopicIdOpt.isEmpty || logTopicIdOpt.contains(requestTopicId)`